src/metadata/tiff.cpp

```diff
@@ -823,15 +823,15 @@ int LibRaw::parse_tiff_ifd(int base)
       FORC3 cam_mul[c] = getreal(11);
       fseek(ifp, 114, SEEK_CUR);
       flip = (get2() >> 7) * 90;
-      if (width * height * 6 == ima_len)
+      if (unsigned(width) * unsigned(height) == ima_len / 6)
```

Comment:
It's hard to say which one is better, but this isn't equal transformation. I guess it would be better to use 64bit types and keep multiply instead of division.

Reply:
Original code is correct, while corrected will succeed even with incorrect ima_len due to integer division rounding.
In real case (correct input from some specific camera(s) ) original code works too w/o changes.